### PR TITLE
Add condition for filter policy events

### PR DESCRIPTION
**Pull request by shanakama**
### Purpose

There will be a new event for health check of GA ( based of event receiving count). No need any processes for health check event, so added another block for filtering POLICY events  for ignore the health check events.

### Issues

https://github.com/wso2/product-microgateway/issues/2747

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/messaging/notification_listener.go [hunk 1]
```diff
@@ -126,8 +127,10 @@ func processNotificationEvent(conf *config.Config, notification *msg.EventNotifi
 		handleApplicationEvents(decodedByte, eventType)
 	} else if strings.Contains(eventType, subscriptionEventType) {
 		handleSubscriptionEvents(decodedByte, eventType)
-	} else {
+	} else if strings.Contains(eventType, policyEventType) {
 		handlePolicyEvents(decodedByte, eventType)
+	} else {
+		// filter all other events including HEALTH_CHECK event
```
Comment (VirajSalaka) [inline adapter/internal/messaging/notification_listener.go:133]: We do not need an else block to keep a comment. Shall we remove that?

Comment (shanakama) [reply adapter/internal/messaging/notification_listener.go]: resolved via : https://github.com/wso2/product-microgateway/pull/2731/commits/8d45d46298b6216131e2be9154d8cc1778eaebc5

File: adapter/internal/messaging/notification_listener.go [hunk 2]
```diff
@@ -126,9 +127,11 @@ func processNotificationEvent(conf *config.Config, notification *msg.EventNotifi
 		handleApplicationEvents(decodedByte, eventType)
 	} else if strings.Contains(eventType, subscriptionEventType) {
 		handleSubscriptionEvents(decodedByte, eventType)
-	} else {
+	} else if strings.Contains(eventType, policyEventType) {
 		handlePolicyEvents(decodedByte, eventType)
 	}
+	// othere events will ignore including HEALTH_CHECK event
```
Comment (VirajSalaka) [inline adapter/internal/messaging/notification_listener.go:133]:
```suggestion
	// other events will ignore including HEALTH_CHECK event
```

Comment (shanakama) [reply adapter/internal/messaging/notification_listener.go]: Resolved via : https://github.com/wso2/product-microgateway/pull/2731/commits/84ed91a49911d68e79cf89036b658ad45cb3b654

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**